portal/v2/src/ClusterDetailList.tsx

```diff
@@ -70,6 +73,32 @@ export class ClusterDetailComponent extends Component<ClusterDetailComponentProp
             <MachineSetsWrapper currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded}/>
           );
         }
+        case "apistatistics":
+        {
+          return (
+            <Statistics currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded} statisticsType={"api"}/>
+          );
+        }
+        case "kcmstatistics":
+          {
+            return (
+              <Statistics currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded} statisticsType={"kcm"}/>
+            );
+          }
+        case "dnsstatistics":
+        {
+          return (
+            <Statistics currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded} statisticsType={"dns"}/>
+          );
+        }
+        case "ingressstatistics":
+        {
+          return (
+            <Statistics currentCluster={this.props.cluster!} detailPanelSelected={this.props.detailPanelVisible} loaded={this.props.isDataLoaded} statisticsType={"ingress"}/>     
+          );
+        }
```

Comment:
I'd like to see this refactored into a map. See my comment on this PR here: https://github.com/Azure/ARO-RP/pull/2587#discussion_r1144217368

Reply:
Yes, you are right, but since this PR did not initiated the whole switch design, I think it would be better if we correct the whole thing and create another work item to refactor those and have whole another PR just for that for not only this piece of code but any other code as well in admin-portal.

Reply:
If you're introducing new code, it should be following better standards, otherwise we keep creating PR debt. I'd understand if it wasn't your own code.

Reply:
Okay I will change the switch case to a map